vaults/smart-contracts/v3/periphery/Yearn4626Router.md

```diff
@@ -0,0 +1,685 @@
+# Yearn4626Router
+[Git Source](https://github.com/yearn/Yearn-ERC4626-Router/blob/68165774ec8858b43db24620756402def14b7ec1/src/Yearn4626Router.sol)
+
+
+Router that is meant to be used with Yearn V3 vaults and strategies
+for deposits, withdraws and migrations.
+The router was developed from the original router by FEI protocol
+https://github.com/fei-protocol/ERC4626
+The router is designed to be used with permit and multicall for the
+optimal experience.
+NOTE: It is important to never leave tokens in the router at the
+end of a call, otherwise they can be swept by anyone.
+
+
+## State Variables
+### _name
+
+```solidity
+bytes32 private immutable _name;
+```
+
+
+## Functions
+### constructor
+
+
+```solidity
+constructor(string memory _name_, IWETH9 weth) PeripheryPayments(weth);
+```
+
+### name
+
+
+```solidity
+function name() external view returns (string memory);
+```
+
+### depositToVault
+
+deposit `amount` to an ERC4626 vault.
+
+*throws "!minShares" Error.*
+
+
+```solidity
+function depositToVault(IYearn4626 vault, uint256 amount, address to, uint256 minSharesOut)
+    public
+    payable
+    override
+    returns (uint256);
+```
+**Parameters**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`vault`|`IYearn4626`|The ERC4626 vault to deposit assets to.|
+|`amount`|`uint256`|The amount of assets to deposit to `vault`.|
+|`to`|`address`|The destination of ownership shares.|
+|`minSharesOut`|`uint256`|The min amount of `vault` shares received by `to`.|
+
+**Returns**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`<none>`|`uint256`|. the amount of shares received by `to`.|
+
+
+### depositToVault
+
+See [depositToVault](/src/Yearn4626Router.sol/contract.Yearn4626Router.md#deposittovault) in IYearn4626Router.
```

Comment:
this link (and most of the links in this file) link to a non-existent file. Fixing to just go to the heading won't work either since the desired link it to the interface (IYearn4626Router), which doesn't get a docs page.

not sure how to best fix